File: src/main/java/sk/hudak/prco/model/GroupOfProductFindEntity.java
package sk.hudak.prco.model;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.io.Serializable;
import java.util.Objects;

@Entity
@Table(name = "GROUP_PRODUCT")
public class GroupOfProductFindEntity {

    @EmbeddedId
    private GroupOfProductFindEntityId id;

    @Column(name = "GROUP_ID", insertable = false, updatable = false)
    private Long groupId;

    @Column(name = "PRODUCT_ID", insertable = false, updatable = false)
    private Long productId;

    public GroupOfProductFindEntityId getId() {
        return id;
    }

    public void setId(GroupOfProductFindEntityId id) {
        this.id = id;
    }

    public Long getGroupId() {
        return groupId;
    }

    public void setGroupId(Long groupId) {
        this.groupId = groupId;
    }

    public Long getProductId() {
        return productId;
    }

    public void setProductId(Long productId) {
        this.productId = productId;
    }

    @Embeddable
    class GroupOfProductFindEntityId implements Serializable {

        @Column(name = "GROUP_ID")
        private Long groupId;

        @Column(name = "PRODUCT_ID")
        private Long productId;

        public Long getGroupId() {
            return groupId;
        }

        public void setGroupId(Long groupId) {
            this.groupId = groupId;
        }

        public Long getProductId() {
            return productId;
        }

        public void setProductId(Long productId) {
            this.productId = productId;
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;
            GroupOfProductFindEntityId that = (GroupOfProductFindEntityId) o;
            return Objects.equals(groupId, that.groupId) &&
                    Objects.equals(productId, that.productId);
        }

        @Override
        public int hashCode() {
            return Objects.hash(groupId, productId);
        }
    }
}

File: src/main/java/sk/hudak/prco/model/GroupOfProductFindEntity.kt
package sk.hudak.prco.model

import lombok.EqualsAndHashCode
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "GROUP_PRODUCT")
class GroupOfProductFindEntity {

    @EmbeddedId
    private var id: GroupOfProductFindEntityId? = null

    @Column(name = "GROUP_ID", insertable = false, updatable = false)
    var groupId: Long? = null

    @Column(name = "PRODUCT_ID", insertable = false, updatable = false)
    var productId: Long? = null

    @EqualsAndHashCode
    @Embeddable
    internal inner class GroupOfProductFindEntityId : Serializable {

        @Column(name = "GROUP_ID")
        var groupId: Long? = null

        @Column(name = "PRODUCT_ID")
        var productId: Long? = null
    }
}
